File: app/src/main/java/com/bkjcb/rqapplication/fragment/BaseSimpleFragment.java
package com.bkjcb.rqapplication.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by DengShuai on 2019/12/30.
 * Description :
 */
public abstract class BaseSimpleFragment extends BaseFragment {
    protected View view;
    protected Context context;
    protected int resId;
    protected Unbinder unbinder;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.context = getContext();
        setResId();
        view = View.inflate(getActivity(), resId, null);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        unbinder = ButterKnife.bind(this, view);
        initView();
        initData();
        return view;
    }

    public abstract void setResId();

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    protected abstract void initView();

    protected abstract void initData();

}

File: app/src/main/java/com/bkjcb/rqapplication/fragment/BaseSimpleFragment.kt
package com.bkjcb.rqapplication.fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by DengShuai on 2019/12/30.
 * Description :
 */
abstract class BaseSimpleFragment : BaseFragment() {
    var contentView: View? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (view == null) {
            contentView = View.inflate(activity, initResID(), null)
        }
        initView()
        initData()
        return view
    }

    abstract fun initResID(): Int
    open fun initView() {}
    open fun initData() {}
}